lib/modules/datasource/maven/index.spec.ts

```diff
@@ -432,6 +441,82 @@ describe('modules/datasource/maven/index', () => {
     expect(res).toMatchSnapshot();
   });
 
+  it('supports artifactregistry urls with auth', async () => {
+    const metadataPaths = [
+      '/org/example/package/maven-metadata.xml',
+      '/org/example/package/1.0.3-SNAPSHOT/maven-metadata.xml',
+      '/org/example/package/1.0.4-SNAPSHOT/maven-metadata.xml',
+      '/org/example/package/1.0.5-SNAPSHOT/maven-metadata.xml',
+    ];
+    const pomfilePath = '/org/example/package/2.0.0/package-2.0.0.pom';
+    hostRules.clear();
+
+    metadataPaths.forEach((path) => {
```

Comment:
Prefer `for (... of someArr)` over `forEach` (here and below), also see [best practices](https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#iterating-objects--containers).